### PR TITLE
Only award reviewer reputation to org members

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/commands/github/organization_member/sync_members.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+module Github
+  class OrganizationMember
+    class SyncMembers
+      include Mandate
+
+      def call
```
Comment (ErikSchierboom) [inline app/commands/github/organization_member/sync_members.rb:6]: I've opted to not create a separate job to handle the logic, which is different from the other two webhooks. The reason for that is simple: this processing is fast enough to be processed by GitHub in a time that is acceptable (green check mark on delivery). Let me know if you'd like the additional job layer instead.

File: app/commands/github/organization_member/sync_members.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              remaining
+              resetAt
+            }
+          }
+        QUERY
+
+        Exercism.octokit_client.post("https://api.github.com/graphql", { query: query }.to_json).to_h
+      end
+
+      def handle_rate_limit(rate_limit_data)
```
Comment (ErikSchierboom) [inline app/commands/github/organization_member/sync_members.rb:56]: At some point I could look into extracting the graphql logic, but I'm not against having it be duplicated for now.

File: app/commands/webhooks/process_membership_update.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+module Webhooks
+  class ProcessMembershipUpdate
+    include Mandate
+
+    initialize_with :action, :username, :organization
+
+    def call
+      return unless organization == 'exercism'
```
Comment (ErikSchierboom) [inline app/commands/webhooks/process_membership_update.rb:8]: This is an extra sanity check. It should not be possible to get here due to us validating the signature of the webhook call, which is only correct if people have access to the secret that we use. If they somehow _do_ get access to the secret, they still wouldn't be able to add a user from another organization.

File: app/commands/github/organization_member/sync_members.rb (hunk 2 above)
Comment (iHiD) [reply app/commands/github/organization_member/sync_members.rb:56]: Yeah. Agreed.

File: app/commands/user/reputation_token/award_for_pull_request_reviewers.rb [hunk 4]
```diff
@@ -14,7 +14,12 @@ def call
           map { |reviewer| reviewer[:reviewer_username] }.
           compact.
           uniq
-        reviewer_usernames.delete(params[:author_username]) # Don't award reviewer reputation to the PR author
+
+        # Don't award reviewer reputation to the PR author
+        reviewer_usernames.delete(params[:author_username])
+
+        # Only award reviewer reputation to organization members
+        reviewer_usernames &= ::Github::OrganizationMember.pluck(:username)
```
Comment (iHiD) [inline app/commands/user/reputation_token/award_for_pull_request_reviewers.rb:22]:
```suggestion
        reviewer_usernames += ::Github::OrganizationMember.pluck(:username)
```

(I've never seen `&=` before)

File: app/commands/webhooks/process_membership_update.rb (hunk 3 above)
Comment (iHiD) [reply app/commands/webhooks/process_membership_update.rb:8]: Nice. I like this thinking!

File: app/controllers/webhooks/organization_updates_controller.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        # params[:action] does not work as it is populated by Rails with the action method name
+        request.request_parameters[:action],
+        params[:membership][:user][:login],
+        params[:organization][:login]
+      )
+    end
+
+    def member_added_or_removed?
+      request.request_parameters[:action] == 'member_added' ||
+        request.request_parameters[:action] == 'member_removed'
```
Comment (iHiD) [inline app/controllers/webhooks/organization_updates_controller.rb]:
```suggestion
      %w{member_added member_removed}.includes?(request.request_parameters[:action])
```

File: app/commands/user/reputation_token/award_for_pull_request_reviewers.rb (hunk 4 above)
Comment (ErikSchierboom) [reply app/commands/user/reputation_token/award_for_pull_request_reviewers.rb:22]: That's not the same I think. `&=` was introduced in Ruby 2.6 and does an intersection.

File: app/commands/github/organization_member/sync_members.rb (hunk 1 above)
Comment (iHiD) [reply app/commands/github/organization_member/sync_members.rb:6]: Is this comment in the wrong place? (ie this code isn't run as part of the webhook as far as I can tell)

File: app/commands/user/reputation_token/award_for_pull_request_reviewers.rb (hunk 4 above)
Comment (iHiD) [reply app/commands/user/reputation_token/award_for_pull_request_reviewers.rb:22]: Oh I see. That's weird but cool.

File: app/commands/webhooks/process_membership_update.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      when 'member_added'
+        add_member!
+      when 'member_removed'
+        remove_member!
+      end
+    end
+
+    private
+    def add_member!
+      ::Github::OrganizationMember.create(username: username)
```
Comment (iHiD) [inline app/commands/webhooks/process_membership_update.rb]:
```suggestion
      ::Github::OrganizationMember.create!(username: username)
```

File: app/commands/webhooks/process_membership_update.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    def remove_member!
+      ::Github::OrganizationMember.destroy_by(username: username)
```
Comment (iHiD) [inline app/commands/webhooks/process_membership_update.rb]:
```suggestion
      ::Github::OrganizationMember.destroy_by!(username: username)
```

Comment (iHiD) [reply app/commands/webhooks/process_membership_update.rb]:
```suggestion
      ::Github::OrganizationMember.where(username: username).update_all(alumni: true)
```

File: app/commands/github/organization_member/sync_members.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/commands/github/organization_member/sync_members.rb:6]: It is 🤦

File: lib/tasks/sync_organization_members.rake [hunk 8]
```diff
@@ -0,0 +1,4 @@
+desc 'Sync organization members'
+task sync_organization_members: :environment do
+  Github::OrganizationMember::SyncMembers.call
```
Comment (ErikSchierboom) [inline lib/tasks/sync_organization_members.rake]:
```suggestion
  Github::OrganizationMember::SyncMembers.()
```

**Pull request merged**